app/src/main/java/com/github/sdp_begreen/begreen/utils/TinyDB.kt

```diff
@@ -0,0 +1,106 @@
+package com.github.sdp_begreen.begreen.utils
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import android.text.TextUtils
+import com.google.gson.Gson
+import java.util.*
+
+// This class is a utility to simplify the use of SharedPreferences for Android.
+class TinyDB(appContext: Context?) {
+
+    /// Instance of SharedPreferences
+    private val preferences: SharedPreferences
+
+    // Initialization of the SharedPreferences object
+    init {
+        preferences = PreferenceManager.getDefaultSharedPreferences(appContext)
+    }
+
+    /**
+     * Get String value from SharedPreferences at 'key'. If key not found, return ""
+     * @param key SharedPreferences key
+     * @return String value at 'key' or "" (empty String) if key not found
+     */
+    fun getString(key: String?): String? {
+        return preferences.getString(key, "")
+    }
+
+    /**
+     * Get parsed ArrayList of String from SharedPreferences at 'key'
+     * @param key SharedPreferences key
+     * @return ArrayList of String
+     */
+    fun getListString(key: String?): ArrayList<String> {
+        return ArrayList(Arrays.asList(*TextUtils.split(preferences.getString(key, ""), "‚‗‚")))
+    }
+
+    /**
+     * Get parsed ArrayList of Objects from SharedPreferences at 'key'
+     * @param key SharedPreferences key
+     * @param mClass a nullable class object
+     * @return ArrayList of String
+     */
+    fun getListObject(key: String?, mClass: Class<*>?): ArrayList<Any> {
```

Comment:
Indeed, there is no need to make mClass nullable. While your proposed return type of List<String> is immutable and might prevent weird behavior, due to time restriction and the good behavior of the current return type, I will leave it as it is. Otherwise, changes made as proposed, thanks a lot!